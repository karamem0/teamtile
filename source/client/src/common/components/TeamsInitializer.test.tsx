//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Presenter from './TeamsInitializer.presenter';

it('should create a shapshot when the loading parameter is true', () => {
  const params = {
    children: (
      <div data-testid="test-Children" />
    ),
    loading: true
  };
  const { asFragment } = render(
    <Presenter {...params} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.queryByTestId('test-Children')).not.toBeInTheDocument();
});

it('should create a shapshot when the loading parameter is false', () => {
  const params = {
    children: (
      <div data-testid="test-Children" />
    ),
    loading: false
  };
  const { asFragment } = render(
    <Presenter {...params} />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTestId('test-Children')).toBeInTheDocument();
});
