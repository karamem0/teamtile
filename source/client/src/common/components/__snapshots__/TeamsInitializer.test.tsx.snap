// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create a shapshot when the loading parameter is false 1`] = `
<DocumentFragment>
  <div
    data-testid="test-Children"
  />
</DocumentFragment>
`;

exports[`should create a shapshot when the loading parameter is true 1`] = `<DocumentFragment />`;
